CMakeLists.txt

```diff
@@ -705,6 +703,10 @@ if(BUILD_TESTING)
   # Add minimal googletest targets. The provided one has many side-effects, and
   # googletest has a very straightforward build.
   add_library(boringssl_gtest third_party/googletest/src/gtest-all.cc)
+  if(BUILD_SHARED_LIBS)
+    # This is needed for the Windows build to correctly annotate GTest's API with __declspec(dllexport)
+    target_compile_options(boringssl_gtest PRIVATE -DGTEST_CREATE_SHARED_LIBRARY=1)
+  endif()
```

Comment:
```suggestion
  if(BUILD_SHARED_LIBS AND MSVC)
    # This is needed for the Windows build to correctly annotate GTest's API with __declspec(dllexport)
    target_compile_options(boringssl_gtest PRIVATE -DGTEST_CREATE_SHARED_LIBRARY=1)
  endif()
```

Reply:
Nit: maybe we can avoid removing the global definition(and add the definition in place) by only call `remove_definitions` inside the if condition because the test seems the final one being built.
https://cmake.org/cmake/help/v3.1/command/remove_definitions.html

Reply:
I'll experiment, but I'm not sure that will work since the tests are in the same folder as some of the source code:
> Removes flags (added by add_definitions) from the compiler command line for sources in the current directory and below.

Reply:
you are right. the test is built before other directories. the function may not work.